Minor spelling corrections and fix of the claimed outcome of the examples in the section on repeated parameters

roberthoedicke opened the pull request.

roberthoedicke committed:
Update 02-identifiers-names-and-scopes.md

Fixed typo.

roberthoedicke committed:
Update 03-types.md

Fixed typo.

roberthoedicke committed:
Update 03-types.md

Inserted missing word "bounds".

roberthoedicke committed:
Update 04-basic-declarations-and-definitions.md

Corrected the claimed outcome of the example in the section on repeated parameters. The example method sum sums up the _squares_ of the arguments.

roberthoedicke committed:
Update 06-expressions.md

Inserted two missing instances of the word "the".
Corrected "invokeDynamic" to "applyDynamic".

roberthoedicke committed:
Update 08-pattern-matching.md

Elided superfluous "a".
Corrected "no" to "not".

roberthoedicke committed:
Update 10-xml-expressions-and-patterns.md

Changed two wrong plurals to singulars, and inserted a comma in an enumeration of alternatives before the last "or".

roberthoedicke committed:
Update 12-the-scala-standard-library.md

Changed "in" to "by the".

retronym commented on spec/04-basic-declarations-and-definitions.md:
```diff
@@ -726,7 +726,7 @@ def sum(args: Int*) = {
 
 The following applications of this method yield `0`, `1`,
-`6`, in that order.
+`14`, in that order.
```
I think the bug in this example is in `arg * arg`, that should be `arg + arg`.

roberthoedicke replied:
That may be so, but in that case the claimed results would also be wrong, they should then read
0, 2, 12.

retronym replied:
ha! let's call it `product` then :)

roberthoedicke replied:
I'd prefer sum_of_squares.

retronym replied:
Time for some more :coffee: for me.

roberthoedicke replied:
Thinking about it, the better approach may actually be to leave the name of the method and the claimed results as they are, and just change the body of the for loop from

```
result += arg * arg
```

to

```
result += arg
```

samuelgruetter replied:
I'd vote for `result += arg`, too.

roberthoedicke replied:
I changed my fork accordingly. Do I need to issue a new pull request?

retronym commented:
Hi @roberthoedicke,

Thank you for your contribution!  Would you mind signing our CLA?

http://www.typesafe.com/contribute/cla/scala

Thanks!

roberthoedicke commented:
Done.

On 09 Oct 2014, at 00:57, Jason Zaugg notifications@github.com wrote:

> Hi @roberthoedicke,
> 
> Thank you for your contribution! Would you mind signing our CLA?
> 
> http://www.typesafe.com/contribute/cla/scala
> 
> Thanks!
> 
> —
> Reply to this email directly or view it on GitHub.

Robert Hödicke 
Managing Director Software Engineering

Wonderlamp Industries GmbH 
Studio Babelsberg (fx. Center) 
August-Bebel-Straße 26-53 
14482 Potsdam 
Germany

Tel:    +49 331 721 4026
Fax:    +49 331 721 4025
Email:  r.hoedicke@wonderlampindustries.com

Handelsregister/Trade Register: Amtsgericht Potsdam HRB 23168 P; Geschäftsführer/Managing Directors: Alexander Dannenberg, Marius Schwarz, Robert Hödicke; Bankverbindung/Bank: HypoVereinsbank UniCreditbank AG; Kontonr./Account no.: 615396664; BLZ/Routing no.: 100 208 90; IBAN: DE76100208900615396664, BIC: HYVEDEMM488; Steuer Nr./VAT no.: DE270910822

IMPORTANT NOTICE: This email message and any attached files are confidential and may contain proprietary technical information, in particular technical information pertaining to actual or projected patent applications as well as technical information constituting confidential know-how as protected by the governing law. The recipient(s) of this email message must not disclose the information contained therein to third parties without express consent of the sender. If you are not the addressee, any disclosure, reproduction, copying, distribution or other dissemination or use of this communication is strictly prohibited. If you have received this transmission in error, please notify the sender immediately and then delete this e-mail message, in its entirety and with all attachments, from your system.

roberthoedicke committed:
Update 04-basic-declarations-and-definitions.md

Reverted the claimed result values of the example in the section on repeated parameters and changed the code of the example method instead.

retronym commented:
LGTM. The build errors are spurious. Thanks for your attention to detail!

retronym commented:
Merging now as this only touches the spec.

retronym commented:
PS: next time, I would recommend to create a feature branch for your changes, and submit your pull request from there. That way you can leave your `2.11.x` branch as an exact mirror of `scala/scala#2.11.x`
